src/features/User/Ui/CreateBooking/CreateBooking.tsx

```diff
@@ -0,0 +1,240 @@
+      setNumberOfPersons((prevValue) => prevValue - 1);
+      updateDateRange();
+    }
+  };
+  // ****************************
+  const updateDateRange = () => {
+    const newEndDate = new Date(selectedDateRange.startDate);
+    newEndDate.setDate(newEndDate.getDate() + numberOfPersons);
+
+    setSelectedDateRange({
+      startDate: selectedDateRange.startDate,
+      endDate: newEndDate,
+      key: "selection",
+    });
+  };
+  /********************************************** */
+  // Get All Rooms
+  const [roomDetails, setRoomDetails] = useState([]);
+  const displayRoomsDetails = () => {
+    axios
+      .get(`${userRoomsDetailsUrl}${roomId}`, {
+        headers: requestHeaders,
+      })
+      .then((response) => {
+        setRoomDetails(response?.data?.data?.room);
+        // console.log(response.data.data.room)
+      })
+      .catch((error) => {
+        console.log(error);
+      });
+  };
+  useEffect(() => {
+    displayRoomsDetails();
+  }, []);
+  /********************************************** */
+  const {
+    register,
+    handleSubmit,
+    // setValue,
+    // watch,
+    formState: { errors },
+  } = useForm();
+  // Définissez les types manuellement si nécessaire
+  interface DateRange {
+    startDate: string;
+    endDate: string;
+    // Autres propriétés...
+  }
+
+  interface RoomDetails {
+    _id: string;
+    price: number;
+    // Autres propriétés...
+  }
+
+  interface BookingRequestBody {
+    bookingStartDate: string;
+    bookingEndDate: string;
+    room: string;
+    totalPrice: number;
+  }
+
+  const[bookingDetails, setBookingDetails] = useState({});
+  const[bookingId, setBookingId] = useState(null);
+  const onSubmit = async (data) => {
+    // console.log(data);
+
+    // Vérifiez si selectedDateRange est correctement défini
+    if (selectedDateRange) {
+      // console.log(
+      //   "Valeur de dateRange hors de handleDateChange :",
+      //   selectedDateRange
+      // );
+      // Utilisez l'interface pour déclarer le type de requestBody
+      const requestBody = {
+        startDate: selectedDateRange.startDate
+          ? formatDate(selectedDateRange.startDate)
+          : "",
+        endDate: selectedDateRange.endDate
+          ? formatDate(selectedDateRange.endDate)
+          : "",
+        room: roomDetails?._id || "",
+        totalPrice: roomDetails?.price || 0,
+      };
+      // console.log("requestBody", requestBody);
+      // Envoyer la requête à votre endpoint de création de réservation
+      await axios
+      .post(createBookingUrl, requestBody, {
+  headers: requestHeaders,
+})
+      .then((response) => {
+        // console.log(response);
+        // console.log(response?.data?.data?.booking?._id);
+        // setBookingId(response.data.data.booking._id)
+const bookingId = response.data.data.booking._id
+        console.log(bookingId );
+
+        navigate(`/user/home/booking-details/${bookingId}`);
+      })
+      .catch((error) => {
```

Comment:
We need to provide a feedback to the user, when something goes wrong, for example when trying to access this page as a visitor, I got no feedback that I need to login first and that I am unauthorized.